x-pack/plugins/siem/server/lib/machine_learning/authz.ts

```diff
@@ -0,0 +1,123 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+import { KibanaRequest } from '../../../../../../src/core/server/';
+import { ILicense } from '../../../../licensing/server';
+import { MlPluginSetup } from '../../../../ml/server';
+import { SetupPlugins } from '../../plugin';
+import { MINIMUM_ML_LICENSE } from '../../../common/constants';
+import { hasMlAdminPermissions } from '../../../common/machine_learning/has_ml_admin_permissions';
+import { isMlRule } from '../../../common/machine_learning/helpers';
+import { RuleType } from '../../../common/detection_engine/types';
+import { Validation } from './validation';
+
+export interface MlAuthz {
+  validateRuleType: (type: RuleType) => Promise<Validation>;
+}
+
+/**
+ * Builds ML authz services
+ *
+ * @param license A {@link ILicense} representing the user license
+ * @param ml {@link MlPluginSetup} ML services to fetch ML capabilities
+ * @param request A {@link KibanaRequest} representing the authenticated user
+ *
+ * @returns A {@link MLAuthz} service object
+ */
+export const buildMlAuthz = ({
+  license,
+  ml,
+  request,
+}: {
+  license: ILicense;
+  ml: SetupPlugins['ml'];
+  request: KibanaRequest;
+}): MlAuthz => {
+  let validation: Promise<Validation> | null = null;
+
+  const validateRuleType = async (type: RuleType): Promise<Validation> => {
+    if (!isMlRule(type)) {
+      return { valid: true, message: undefined };
+    } else {
+      if (!validation) {
+        validation = validateMlAuthz({ license, ml, request });
+      }
+      return validation;
+    }
+  };
+
+  return { validateRuleType };
+};
+
+/**
+ * Validates ML authorization for the current request
+ *
+ * @param license A {@link ILicense} representing the user license
+ * @param ml {@link MlPluginSetup} ML services to fetch ML capabilities
+ * @param request A {@link KibanaRequest} representing the authenticated user
+ *
+ * @returns A {@link Validation} validation
+ */
+export const validateMlAuthz = async ({
+  license,
+  ml,
+  request,
+}: {
+  license: ILicense;
+  ml: SetupPlugins['ml'];
+  request: KibanaRequest;
+}): Promise<Validation> => {
+  let message: string | undefined;
+
+  if (ml == null) {
+    message = i18n.translate('xpack.siem.authz.mlUnavailable', {
+      defaultMessage: 'The machine learning plugin is not available. Try enabling the plugin.',
+    });
+  } else if (!hasMlLicense(license)) {
+    message = i18n.translate('xpack.siem.licensing.unsupportedMachineLearningMessage', {
+      defaultMessage:
+        'Your license does not support machine learning. Please upgrade your license.',
+    });
+  } else if (!(await isMlAdmin({ ml, request }))) {
+    message = i18n.translate('xpack.siem.authz.userIsNotMlAdminMessage', {
+      defaultMessage: 'The current user is not a machine learning administrator.',
+    });
+  }
+
+  return {
+    valid: !message,
```

Comment:
FYI, if for some reason down the road a translation returns an empty string for whatever reason you are going to get that the user is a valid machine learning user if there is a mistake in translations at some point since:

```ts
!''
true
```